sklearn/neural_network/extreme_learning_machines.py

```diff
@@ -0,0 +1,629 @@
+        if self.n_hidden <= 0:
+            raise ValueError("n_hidden must be > 0, got %s." % self.n_hidden)
+        if self.C <= 0.0:
+            raise ValueError("C must be > 0, got %s." % self.C)
+        if self.activation not in ACTIVATIONS:
+            raise ValueError("The activation %s is not supported. Supported "
+                             "activation are %s." % (self.activation,
+                                                     ACTIVATIONS))
+
+        # Initialize public attributes
+        if not hasattr(self, 'classes_'):
+            self.classes_ = None
+        if not hasattr(self, 'coef_hidden_'):
+            self.coef_hidden_ = None
+
+        # Initialize private attributes
+        if not hasattr(self, '_HT_H_accumulated'):
+            self._HT_H_accumulated = None
+
+        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'],
+                         dtype=np.float64, order="C", multi_output=True)
+
+        # This outputs a warning when a 1d array is expected
+        if y.ndim == 2 and y.shape[1] == 1:
+            y = column_or_1d(y, warn=True)
+
+        # Classification
+        if isinstance(self, ClassifierMixin):
+            self.label_binarizer_.fit(y)
+
+            if self.classes_ is None or not incremental:
+                self.classes_ = self.label_binarizer_.classes_
+                if sample_weight is None:
+                    sample_weight = compute_sample_weight(self.class_weight,
+                                                          self.classes_, y)
+            else:
+                classes = self.label_binarizer_.classes_
+                if not np.all(np.in1d(classes, self.classes_)):
+                    raise ValueError("`y` has classes not in `self.classes_`."
+                                     " `self.classes_` has %s. 'y' has %s." %
+                                     (self.classes_, classes))
+
+            y = self.label_binarizer_.transform(y)
+
+        # Ensure y is 2D
+        if y.ndim == 1:
+            y = np.reshape(y, (-1, 1))
+
+        n_samples, n_features = X.shape
+        self.n_outputs_ = y.shape[1]
+
+        # Step (1/2): Compute the hidden layer coefficients
+        if (self.coef_hidden_ is None or (not incremental and
+                                          not self.warm_start)):
+            # Randomize and scale the input-to-hidden coefficients
+            self._init_weights(n_features)
+
+        # Step (2/2): Compute hidden-to-output coefficients
+        if self.batch_size is None:
+            # Run the least-square algorithm on the whole dataset
+            batch_size = n_samples
+        else:
+            # Run the recursive least-square algorithm on mini-batches
+            batch_size = self.batch_size
+
+        batches = gen_batches(n_samples, batch_size)
+
+        # (First time call) Run the least-square algorithm on batch 0
+        if not incremental or self._HT_H_accumulated is None:
```

Comment:
There seems a lot of repetition between what happens in the first batch and the other batches. Have you tried to refactor that?